### ISSUE TITLE
How to render a number in a specific number system

### ISSUE THREAD

**Issue by rotsee** [papandreou/node-cldr#71]
I'm probably missing something, but I can't seem to figure out how to render a number in a specific number system. To render a number, I need a number format, a set of number symbols, and a set of digits. Getting the former is straightforward, but `extractDigitsByNumberSystemId()` sometimes return the name of a RBNF method, _that is dependent on the locale_. How am I then supposed to know what locale to use?

This is unproblematic:
```
> cldr.extractDigitsByNumberSystemId().ethi
'renderEthiopic'
> cldr.extractRbnfFunctionByType().renderEthiopic(1234)  // The locale doesn't matter here
'፲፪፻፴፬'
```

This, however, is trickier:
```
> cldr.extractDigitsByNumberSystemId().jpan
'renderSpelloutCardinal'
> cldr.extractRbnfFunctionByType('sv').renderSpelloutCardinal(1234)  // This won't do it
'1234'
> cldr.extractRbnfFunctionByType('ja').renderSpelloutCardinal(1234)
'千二百三十四'
```
How can I determine what language/locales (”ja” in this case) I should use with the method returned by `extractDigitsByNumberSystemId()`?

My current work-around looks something like this:
```
> let digits = cldr.extractDigitsByNumberSystemId()[numberSystem]
> let res = cldr.extractRbnfFunctionByType(locale).renderSpelloutCardinal(number)
> if (res === "" + number) { throw new Error(`${numberSystem} is not a supported number system for ${locale}.`) }
```

**Comment by papandreou** [comment]
Hmm, it seems like you've found a bug. The source data for `extractDigitsByNumberSystemId` is https://github.com/papandreou/node-cldr/blob/master/3rdparty/cldr/common/supplemental/numberingSystems.xml. Some of the `algorithmic` ones do say which locale to use: https://github.com/papandreou/node-cldr/blob/master/3rdparty/cldr/common/supplemental/numberingSystems.xml#L43

But that information is thrown away by `extractDigitsByNumberSystemId` here where it just takes the last fragment: https://github.com/papandreou/node-cldr/blob/7b2dec87883a39127c4b8e05fc531c9fd23682b8/lib/cldr.js#L1513

... thus discarding the `ja/SpelloutRules/` part that's actually significant here. We should see about exposing that somehow. What would be the most practical for you?

The `algorithmic` rules that *don't* specify a locale tend to be the ones that don't vary between the locales, where all locales effectively inherit the one defined in the `root` locale: https://github.com/papandreou/node-cldr/blob/master/3rdparty/cldr/common/rbnf/root.xml

For all of those I think the answer is that you can just use any locale, or just `cldr.extractRbnfFunctionByType('root')`

**Comment by papandreou** [comment]
Maybe we should just rip off the band-aid and do a breaking change so that the return value is more one-to-one with what's in the data:

```js
cldr.extractDigitsByNumberSystemId().ethi
// { type: 'algorithmic', rules: 'ethiopic' }

cldr.extractDigitsByNumberSystemId().jpan
// { type: 'algorithmic', rules: 'spelloutCardinal', locale: 'ja' }
```

**Comment by papandreou** [comment]
@rotsee, would #72 do the trick?

**Comment by rotsee** [comment]
This certainly works for me (much cleaner and nicer, from a user perspective!), if you are happy with changing the behavior like that.

Note that a typo seems to have sneaked into the old method name: `extracingtDigitsByNumberSystemId`: https://github.com/papandreou/node-cldr/pull/72/commits/487ea0351e5ab952bc8dc8a8575a0fd90d576240#diff-414150d1f555b27bf10d664a622d2d33R1530

**Comment by papandreou** [comment]
Ah, nice catch. Fixed!

Released the change in 4.14.0

**Comment by papandreou** [comment]
Added docs here: https://github.com/papandreou/node-cldr#cldrextractnumberingsystemnumberingsystemid
